app/src/main/java/com/yacgroup/yacguide/database/tourbook/JsonExporter.kt

```diff
@@ -17,50 +17,39 @@
 
 package com.yacgroup.yacguide.database.tourbook
 
-import android.content.ContentResolver
 import android.net.Uri
 import com.yacgroup.yacguide.database.Ascend
-import com.yacgroup.yacguide.database.DatabaseWrapper
 import org.json.JSONArray
 import org.json.JSONException
 import org.json.JSONObject
 import java.io.IOException
 
-class JsonExporter(
-    private val _db: DatabaseWrapper,
-    _contentResolver: ContentResolver): BaseExporter(_contentResolver) {
-
-    private val JSON_INDENT_SPACE = 2
+open class JsonExporter: BaseExporter() {
+    companion object {
+        const val JSON_INDENT_SPACE = 2
+    }
 
     @Throws(IOException::class)
```

Comment:
At the moment, only the IO exception is catched at this point. However, method `_ascend2Json` also can throw an JSON exception. Do we want to map all these exceptions to one project specific Exception and handle it than in the activity? What is a good practice there? From my point of view it makes no sense to make alle these low-level exceptions visible in the activity.

Reply:
JSON Exception inherits from IOException, so this should not be a problem.. However, you are right regarding unified exception handling. If you already have any ideas, you may create a ticket.

Reply:
I don't have a specific idea but I'm going to create a ticket as base for discussions.